Applications/src/auto-contrast.cc

```diff
@@ -0,0 +1,104 @@
+/*=========================================================================
+
+  Library   : Image Registration Toolkit (IRTK)
+  Module    : $Id$
+  Copyright : Imperial College, Department of Computing
+              Visual Information Processing (VIP), 2008 onwards
+  Date      : $Date$
+  Version   : $Revision$
+  Changes   : $Author$
+Copyright (c) 1999-2014 and onwards, Imperial College London
+All rights reserved.
+See LICENSE for details
+
+=========================================================================*/
+
+#include "mirtk/Common.h"
+#include "mirtk/Options.h"
+
+#include "mirtk/ImageReader.h"
+#include "mirtk/IOConfig.h"
+#include "mirtk/GenericImage.h"
+
+using namespace mirtk;
+
+char *output_name = NULL;
+
+
+void PrintHelp(const char *name)
+{
+  cout << endl;
+  cout << "Usage: " << name << " <image> <output> [options]" << endl;
+  cout << endl;
+  cout << "Description:" << endl;
+  cout << "  This program auto contrast an image." << endl;
+  cout << endl;
+  cout << "Arguments:" << endl;
+  cout << "  input    Input image." << endl;
+  cout << "  output   Contrasted output image." << endl;
+  cout << endl;
+  cout << "Options:" << endl;
+  cout << "  -Tp <value>        Target padding value all pixel < value = value" << endl;
+  cout << "  -Tp2 <maxvalue>    Target padding maximum value all pixel > maxvalue = maxvalue" << endl;
+  PrintCommonOptions(cout);
+  cout << endl;
+}
+
+int main(int argc, char *argv[])
+{
+  // Parse positional arguments
+  REQUIRES_POSARGS(2);
+  // Determine input image data type
+  InitializeIOLibrary();
```

Comment:
I would initialize the IO library and read the image data after the option parsing. If `-h` or `-help` option is given, the program won't need to read the input data.